src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -360,26 +365,45 @@ static void uct_rdmacm_cm_handle_error_event(struct rdma_cm_event *event)
     char ip_port_str[UCS_SOCKADDR_STRING_LEN];
     char ep_str[UCT_RDMACM_EP_STRING_LEN];
     uct_cm_remote_data_t remote_data;
+    uct_rdmacm_priv_data_hdr_t *hdr;
     ucs_log_level_t log_level;
     ucs_status_t status;
 
-    if (event->event == RDMA_CM_EVENT_REJECTED) {
+    switch (event->event) {
+    case RDMA_CM_EVENT_REJECTED:
         if (cep->flags & UCT_RDMACM_CM_EP_ON_SERVER) {
             /* response was rejected by the client in the middle of
              * connection establishment, so report connection reset */
             status = UCS_ERR_CONNECTION_RESET;
         } else {
             ucs_assert(cep->flags & UCT_RDMACM_CM_EP_ON_CLIENT);
-            status = UCS_ERR_REJECTED;
+            hdr = (uct_rdmacm_priv_data_hdr_t *)event->param.conn.private_data;
+
+            if ((hdr != NULL) && (event->param.conn.private_data_len > 0) &&
+                ((ucs_status_t)hdr->status == UCS_ERR_REJECTED)) {
+                ucs_assert(hdr->length == 0);
+                ucs_assert(event->param.conn.private_data_len >= sizeof(*hdr));
```

Comment:
taking into account the assert above, shout not `event->param.conn.private_data_len == sizeof(*hdr)` ?

Reply:
rdmacm adds some additional info into this value. 
the value of event->param.conn.private_data_len isn't the same as it was set when calling rdma_reject.

from rdma_reject man page:
```
private_data_len
                   Specifies  the  size  of the user-controlled data buffer.  Note that the actual amount of data transferred to the remote side is transport dependent and may be larger than that
                   requested.
```

Reply:
ic, 10x for clarification, maybe add a comment?

Reply:
sure, will add a comment.